code/modules/mob/living/carbon/alien/damage_procs.dm

```diff
@@ -1,21 +1,26 @@
 
+/mob/living/carbon/alien/apply_damage(damage = 0,damagetype = BRUTE, def_zone = null, blocked = 0)
+	var/hit_percent = (100-blocked)/100
+	if(!damage || (hit_percent <= 0))
+		return 0
+	switch(damagetype)
+		if(BRUTE)
+			adjustBruteLoss(damage * hit_percent)
+		if(BURN)
+			adjustFireLoss(damage * 2 * hit_percent) // Xenos take double burn damage, MUY IMPORTANTE
+		if(OXY)
+			if(damage < 0) //we shouldn't be taking oxygen damage through this proc, but we'll let it heal.
+				adjustOxyLoss(damage * hit_percent)
+	return 1
+
+//aliens are immune to stamina and toxin damage.
+/mob/living/carbon/alien/adjustStaminaLoss(amount, updating_stamina = 1)
+	return
 
+/mob/living/carbon/alien/setStaminaLoss(amount, updating_stamina = 1)
+	return
+	
 /mob/living/carbon/alien/getToxLoss()
 	return 0
 
 /mob/living/carbon/alien/adjustToxLoss(amount, updating_health = TRUE, forced = FALSE) //alien immune to tox damage
-	return FALSE
```

Comment:
I am not sure, but I fear having an empty proc declaration without a return may cause BYOND to treat it as if it were not an override. There has to be a reason nearly all procs written this way have `return` in them. Without the `return`, I think it will cause BYOND to use the /carbon `adjustToxLoss()` parent, removing the tox immunity. I am asking @Robustin to test this and make sure he did not break it.